### PR TITLE
added libpq port

### PR DESCRIPTION
**Pull request by barcharcraz**
I wrote a cmake based build system for libpq. The exports seem to be correct but I still need to test it. Still I'm adding the PR now so others can have a look and spot any glaring errors in my cmake code.

I didn't install exports because the findPostgreSQL module that ships with cmake does not use exported targets.

This should enable us to add the postgres plugin for Qt, as well as odb-pq.

#113 also may have been blocked on a libpq port but I'm not sure (maybe it implements the database protocol itself).

### CODE REVIEW

File: ports/libpq/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    src/interfaces/libpq/libpq-fe.h
+    )
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+add_library(libpq ${pg_port_src} ${pg_backend_src} ${pg_libpq_src})
+target_compile_definitions(libpq PRIVATE -DFRONTEND -DENABLE_THREAD_SAFETY -DUSE_OPENSSL -D_CRT_SECURE_NO_WARNINGS)
+target_link_libraries(libpq PRIVATE OpenSSL::SSL ws2_32 secur32 advapi32 shell32)
+target_include_directories(libpq PRIVATE src/include src/include/port/win32 src/include/port/win32_msvc src/port)
+
+install(TARGETS libpq
+    ARCHIVE DESTINATION lib
```
Comment (Orvid) [inline ports/libpq/CMakeLists.txt]: Basically every other project, and the CMake docs themselves, do these in the order `RUNTIME`, `LIBRARY`, `ARCHIVE`.

File: ports/libpq/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    LIBRARY DESTINATION lib
```
Comment (Orvid) [inline ports/libpq/CMakeLists.txt]: As we have control, install libraries (`.dll`s) directly to `bin`.

Comment (barcharcraz) [inline ports/libpq/CMakeLists.txt]: That's what this does. Runtime destination is special sauce that does this on windows

**Comment by ras0219-msft** [comment]
This looks pretty good to me, let me know when you're done tweaking and ready for it to be merged.

File: ports/libpq/CMakeLists.txt (hunk 2 above)
Comment (Orvid) [inline ports/libpq/CMakeLists.txt]: Ah, didn't realize it did that.

**Comment by barcharcraz** [comment]
This should be good to go. I've tested it for one of our internal projects and it seems to work fine.

**Comment by ras0219-msft** [comment]
Awesome, looks good for me too. Great work, both of you!

**Pull request merged**